kotest-assertions/kotest-assertions-json/src/jvmMain/kotlin/io/kotest/assertions/json/JsonCountableElement.kt

```diff
@@ -0,0 +1,8 @@
+package io.kotest.assertions.json
+
+sealed class JsonCountableElement {
+
+    class JsonKeyValuePairs internal constructor(val count: Int) : JsonCountableElement()
+}
+
+val Int.jsonKeyValueEntries get() = JsonCountableElement.JsonKeyValuePairs(this)
```

Comment:
Are both of these supposed to be public?

Reply:
Do you suggest something like the following?
```kotlin
sealed class JsonCountableElement {

    internal class JsonKeyValuePairs(val count: Int) : JsonCountableElement()
}

val Int.jsonKeyValueEntries: JsonCountableElement get() = JsonCountableElement.JsonKeyValuePairs(this)
```

I have no preference here. If you suggest to restrict visibility of `JsonKeyValuePairs`, I'll do it.

Reply:
I've decided not to introduce this API. It's not intuitive enough. I've described the API design in the documentation in the recent commit.